lib/amazon_ssa_support/rolling_s3_outputter.rb

```diff
@@ -0,0 +1,110 @@
+require "log4r/outputter/fileoutputter"
+require "log4r/staticlogger"
+
+module Log4r
+  
+  class RollingS3Outputter < FileOutputter
+    attr_reader :count, :maxsize
+
+    def initialize(_name, hash={})
+      @count           = 0
+      @aws_args        = hash[:aws_args]
+      @evm_bucket_name = @aws_args[:evm_bucket]
+      @s3_log_prefix   = File.join(@aws_args[:log_prefix], @aws_args[:extractor_id])
+      @evm_bucket      = AmazonSsaSupport::EvmBucket.get(@aws_args)
+            
+      _maxsize = (hash[:maxsize] or hash['maxsize']).to_i
+      if _maxsize.class != Fixnum
```

Comment:
@hsong-rh that doesn't do the same thing that the Fixnum test was doing.  Checking for Integer makes sure it is a whole number.  Checking for Fixnum checked the size too.  That's why I asked about the size issue.  Do we need to know that the maxsize isn't too big?